server/svix-server/src/expired_message_cleaner.rs

```diff
@@ -14,9 +16,18 @@ pub async fn clean_expired_messages(
     pool: &DatabaseConnection,
     limit: u32,
 ) -> std::result::Result<UpdateResult, DbErr> {
-    let legacy_stmt = Statement::from_sql_and_values(
-        pool.get_database_backend(),
-        r#"
+    //
+    let legacy_res = {
```

Comment:
I know why we're calling this legacy, but maybe it'd be better just to name these after the models they are destroying?

Or fill in the comment above about why this is "legacy"?

Reply:
Can do.


Separately, a question I have for you is: should we add cfg to enable/disable this bit specifically since the worst-case search is inevitable after a certain point for anyone already switched over to messagecontent?

The most cost-effective thing to do is _not run this query_ after it's no longer needed, but I'm not sure how to know this programmatically.

I suppose if there are no rows with future expiration dates we can just skip this entirely since messagecontent would get all the new writes, right?

Reply:
I have done some refactoring and included some docs to try and explain "legacy."

I also did the thing where I check for pending non-null payloads in the `message` table. This seems important to have since otherwise installations will continue to scan that old table forever while never writing new payloads to it. That's a lose-lose situation.